man/systemd.exec.xml

```diff
@@ -3690,6 +3690,15 @@ StandardInputData=V2XigLJyZSBubyBzdHJhbmdlcnMgdG8gbG92ZQpZb3Uga25vdyB0aGUgcnVsZX
           system.</para></listitem>
         </varlistentry>
 
+        <varlistentry>
+          <term><varname>$TRIGGER_REASON</varname></term>
+
+          <listitem><para>If the unit was activated dynamically (e.g.: a corresponding path unit), the reason
+          for the activation will be listed in this variable. Note that this is provided in a best-effort way.
+          For example, there might be multiple triggers firing at any given time, and only one will be reported,
+          and there is no guarantee on which one it will be.</para></listitem>
+        </varlistentry>
+
       </variablelist>
```

Comment:
so, what if I have a unit that can be triggered by both path and timer? I'll get one TRIGGER_REASON and I'll not have a clue how to interpret TRIGGER_REASON in that case.

Reply:
well it's either a path or a foo.timer, should be pretty easy to tell apart no?

Reply:
that would mean you could never add this logic for .socket units, too. i.e. think about a .socket unit that has two Listen* lines for two unix sockets. It might be interesting to propagate via this new infra the info which of the two sockets caused us to activate the service. Yes, the daemon could look at the two fds and guess, if only one of them is busy, but this would never answer you the question: which was triggered first...

So given AF_UNIX sockets are typically in the fs, and thus a path, you'd then create ambiguity too. i.e. were you activated because of an inotify event or because of traffic incoming on the AF_UNIX socket that the path refers to.